website/source/api/acl/roles.html.md

```diff
@@ -0,0 +1,462 @@
+---
+layout: api
+page_title: ACL Roles - HTTP API
+sidebar_current: api-acl-roles
+description: |-
+  The /acl/role endpoints manage Consul's ACL Roles.
+---
+
+-> **1.5.0+:**  The APIs are available in Consul versions 1.5.0 and later. The documentation for the legacy ACL API is [here](/api/acl/legacy.html)
+
+# ACL Role HTTP API
+
+The `/acl/role` endpoints [create](#create-a-role), [read](#read-a-role),
+[update](#update-a-role), [list](#list-roles) and [delete](#delete-a-role)  ACL roles in Consul.
+For more information about ACLs, please see the [ACL Guide](/docs/guides/acl.html).
+
+## Create a Role
+
+This endpoint creates a new ACL role.
+
+| Method | Path                         | Produces                   |
+| ------ | ---------------------------- | -------------------------- |
+| `PUT`  | `/acl/role`                  | `application/json`         |
+
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes),
+[agent caching](/api/index.html#agent-caching), and
+[required ACLs](/api/index.html#acls).
+
+| Blocking Queries | Consistency Modes | Agent Caching | ACL Required |
+| ---------------- | ----------------- | ------------- | ------------ |
+| `NO`             | `none`            | `none`        | `acl:write`  |
+
+### Parameters
+
+- `Name` `(string: <required>)` - Specifies a name for the ACL role. The name
+   can only contain alphanumeric characters as well as `-` and `_` and must be
+   unique. 
+   
+~> TODO(rb): update this.
+
+- `Description` `(string: "")` - Free form human readable description of the role.
+
+- `Policies` `(array<PolicyLink>)` - The list of policies that should
+   be applied to the role. A PolicyLink is an object with an "ID" and/or "Name" field
+   to specify a policy. With the PolicyLink, roles can be linked to policies either by the
+   policy name or by the policy ID. When policies are linked by name they will be
+   internally resolved to the policy ID. With linking roles internally by IDs,
+   Consul enables policy renaming without breaking tokens.
+
+- `ServiceIdentities` `(array<ServiceIdentity>)` - The list of service
```

Comment:
I think I still struggle a little with this concept. Maybe we can link out to the location where we define `service identity`